src/Raven.Studio/typescript/common/helpers/graph/graphHelper.ts

```diff
@@ -105,25 +105,34 @@ class graphHelper {
         return config;
     }
     
-    static drawErrorMark(ctx: CanvasRenderingContext2D, x: number, y: number, dx: number) {
-        const markWidth = 8;
+    static drawErrorMark(ctx: CanvasRenderingContext2D, x: number, y: number, dx: number, dy: number = 8) {
+        const markWidth = dy;
         if (dx > markWidth) {
             // draw full triangle
             ctx.beginPath();
-            ctx.moveTo(x+ dx, y);
+            ctx.moveTo(x + dx, y);
             ctx.lineTo(x + dx - markWidth, y);
             ctx.lineTo(x + dx, y + markWidth);
             ctx.fill();
         } else if (dx > 1) {
             ctx.beginPath();
-            ctx.moveTo(x+ dx, y);
+            ctx.moveTo(x + dx, y);
             ctx.lineTo(x, y);
             ctx.lineTo(x, y + markWidth - dx);
             ctx.lineTo(x + dx, y + markWidth);
             ctx.fill();
         }
     }
     
+    static drawPendingArea(ctx: CanvasRenderingContext2D, x: number, y: number, dx: number) {
+        ctx.setLineDash([8, 2]);
```

Comment:
we do need to reset that before next paint?

Reply:
as discussed, this is resolved with `context.save` & `context.restore` per item in loop